packages/xml-views-validation/src/validators/elements/type-of-aggregation.ts

```diff
@@ -0,0 +1,105 @@
+import { XMLElement } from "@xml-tools/ast";
+import {
+  UI5SemanticModel,
+  UI5Interface,
+  UI5Class,
+  UI5Type,
+} from "@ui5-language-assistant/semantic-model-types";
+import {
+  getUI5ClassByXMLElement,
+  getUI5AggregationByXMLElement,
+  classIsOfType,
+} from "@ui5-language-assistant/logic-utils";
+import { InvalidAggregationTypeIssue } from "../../../api";
+import { INVALID_AGGREGATION_TYPE, getMessage } from "../../utils/messages";
+
+export function validateAggregationType(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): InvalidAggregationTypeIssue[] {
+  const ui5class = getUI5ClassByXMLElement(xmlElement, model);
+  // Root element can never be inside aggregation
+  if (xmlElement.parent.type === "XMLDocument") {
+    return [];
+  }
+
+  const allowedTypeInAggregation = getAllowedTypeForSubNodesByXMLElement(
+    xmlElement.parent,
+    model
+  );
+  if (ui5class !== undefined && allowedTypeInAggregation !== undefined) {
```

Comment:
flip condition and return empty array if we do not have enough data to compute
and at the last statement return the result of getInvalidAggregationTypeIssue